merkle/tests/tree_poseidon.rs

```diff
@@ -0,0 +1,166 @@
+        <TestHasher as MerkleHasher>::Hash,
+        TestHasher,
+        MinaPoseidonMerkleMerger,
+    >;
+
+    // The test case is from genesis ledger
+    #[test]
+    fn test_mina_poseidon_merkle_merger_even() {
+        let h1: Fp = BigInteger256::read(
+            [
+                46, 111, 163, 222, 5, 13, 158, 61, 44, 42, 248, 84, 17, 204, 170, 242, 152, 233,
+                200, 16, 116, 180, 10, 25, 151, 67, 118, 244, 69, 231, 109, 46,
+            ]
+            .as_slice(),
+        )
+        .unwrap()
+        .into();
+        let h2: Fp = BigInteger256::read(
+            [
+                145, 28, 178, 157, 78, 192, 208, 17, 235, 147, 39, 0, 105, 97, 173, 134, 12, 183,
+                159, 53, 66, 254, 56, 82, 142, 108, 173, 182, 106, 220, 226, 6,
+            ]
+            .as_slice(),
+        )
+        .unwrap()
+        .into();
+        let h3: Fp = BigInteger256::read(
+            [
+                146, 128, 118, 98, 200, 74, 91, 121, 164, 107, 160, 240, 39, 236, 147, 102, 92,
+                114, 200, 62, 92, 15, 190, 134, 196, 11, 200, 126, 35, 56, 168, 59,
+            ]
+            .as_slice(),
+        )
+        .unwrap()
+        .into();
+        // Index of the root node is 0
+        let node_index = 0;
+        // Genesis ledger has 1676 accounts
+        const GENESIS_LEDGER_ACCOUNT_NUM: u64 = 1676;
+        let mut tree = TestMerkleTree::new();
+        for i in 0..GENESIS_LEDGER_ACCOUNT_NUM {
+            tree.add(TestLeafNode(i))
+        }
+        let meta = MerkleTreeNodeMetadata::new(node_index, tree.height());
+        let merged =
+            MinaPoseidonMerkleMerger::merge([Some(h1), Some(h2)], meta).unwrap_or_default();
+        assert_eq!(h3, merged);
+    }
+
+    #[test]
+    fn test_mina_poseidon_merkle_merger_odd() {
+        let h1: Fp = BigInteger256::read(
+            [
+                146, 128, 118, 98, 200, 74, 91, 121, 164, 107, 160, 240, 39, 236, 147, 102, 92,
+                114, 200, 62, 92, 15, 190, 134, 196, 11, 200, 126, 35, 56, 168, 59,
+            ]
+            .as_slice(),
+        )
+        .unwrap()
+        .into();
+        let h2: Fp = BigInteger256::read(
+            [
+                181, 78, 220, 55, 96, 63, 185, 35, 72, 225, 102, 60, 60, 81, 94, 219, 35, 27, 49,
+                124, 235, 72, 211, 61, 223, 235, 94, 79, 29, 22, 102, 15,
+            ]
+            .as_slice(),
+        )
+        .unwrap()
+        .into();
+        let meta = MerkleTreeNodeMetadata::new(0, 12);
+        let merged = MinaPoseidonMerkleMerger::merge([Some(h1), None], meta).unwrap_or_default();
+        assert_eq!(h2, merged);
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    #[test]
+    fn genesis_ledger_parity_test() {
+        use rocksdb::*;
+
+        let db =
+            DB::open_for_read_only(&Options::default(), "../ledger/test-data/genesis_ledger_6a887ea130e53b06380a9ab27b327468d28d4ce47515a0cc59759d4a3912f0ef/", true).unwrap();
```

Comment:
I think ideally our DbBackedGenesisLedger should be useable here. This is a lot of messy database logic to include in a test

Reply:
Yeah, I will leave a TODO here